src/payload/checkpoint.rs

```diff
@@ -552,76 +553,215 @@ impl<P: Platform> Display for Checkpoint<P> {
 mod tests {
 	use {
 		crate::{
-			alloy::primitives::U256,
-			prelude::{BlockContext, Ethereum},
-			test_utils::{BlockContextMocked, FundedAccounts, transfer_tx},
+			payload::checkpoint::{Checkpoint, IntoExecutable, Mutation},
+			prelude::*,
+			reth::primitives::Recovered,
+			test_utils::{BlockContextMocked, test_bundle, test_tx, test_txs},
 		},
 		std::time::Instant,
 	};
 
+	/// Helper test function to apply multiple transactions on a checkpoint
+	fn apply_multiple<P: PlatformWithRpcTypes>(
+		root: Checkpoint<P>,
+		txs: &[Recovered<types::Transaction<P>>],
```

Comment:
Maybe `txs: impl Iterator<Item=types::Transaction<P>>>`?

Reply:
sorry, was thinking that this is at the `Checkpoint` type, not in a test helper. Forget everything I said ;-)

Reply:
Yep I'm not sure if it's a good idea to add a `apply_many` at the checkpoint type level but we can discuss this separately